pytroll_collectors/geographic_gatherer.py

```diff
@@ -103,8 +106,10 @@ def _setup_triggers(self):
 
     def run(self):
         """Run granule triggers."""
+        signal.signal(signal.SIGTERM, self._handle_sigterm)
         try:
             while True:
+                self._check_sigterm()
```

Comment:
why not make that function return a boolean and use it in the while statement?

Reply:
Adjusted in https://github.com/pytroll/pytroll-collectors/pull/157/commits/43b8cc73039b87da43c6252dd83bdf5c861a9b97